### ISSUE TITLE
Rule Change: prefer-const should allow use of `let` when overriden inside a try catch

### ISSUE THREAD

**Issue by timvandam** [eslint/eslint#15460]
### What rule do you want to change?

prefer-const

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
let x;
try {
  x = functionThatMightThrow(); // ESLint: 'x' is never reassigned. Use 'const' instead.(prefer-const)
} catch (e) {
  return handleError(e);
}

try {
  doSomethingElse(x);
} catch (e) {
  return handleOtherError(e);
}

return x * x * 2 + 20
```

### What does the rule currently do for this code?

Currently, it says that `x` should be a const since it is only assigned once. This is possible, however that would force these try catch blocks to become nested. This is not very neat, especially if there are many try catches

The extra `let x;` can be avoided by using `var x = functionThatMightThrow();`, but that is also not allowed by this rule

### What will the rule do after it's changed?

It will allow using `let` when setting the variable just once, as long as it is set inside of a try catch (and potentially using `var` inside of try-catch if the variable should be function scoped)

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

I am willing to submit a pull request, if I can figure out how to change this :)

**Comment by mdjermanovic** [comment]
Hi @timvandam, thanks for the issue!

> It will allow using `let` when setting the variable just once, as long as it is set inside of a try catch

This is already the expected behavior. If it doesn't work like that, then it's a bug, but I can't reproduce it. Here's [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1jb25zdDogZXJyb3IgKi9cblxubGV0IHg7XG50cnkge1xuICB4ID0gZnVuY3Rpb25UaGF0TWlnaHRUaHJvdygpOyAvLyBFU0xpbnQ6ICd4JyBpcyBuZXZlciByZWFzc2lnbmVkLiBVc2UgJ2NvbnN0JyBpbnN0ZWFkLihwcmVmZXItY29uc3QpXG59IGNhdGNoIChlKSB7XG4gIHJldHVybiBoYW5kbGVFcnJvcihlKTtcbn1cblxudHJ5IHtcbiAgZG9Tb21ldGhpbmdFbHNlKHgpO1xufSBjYXRjaCAoZSkge1xuICByZXR1cm4gaGFuZGxlT3RoZXJFcnJvcihlKTtcbn1cblxucmV0dXJuIHggKiB4ICogMiArIDIwIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6eyJnbG9iYWxSZXR1cm4iOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) with your example,  and there's no `'x' is never reassigned` error. Can you open a bug report with all the details?

**Comment by timvandam** [comment]
I have recreated it in this repo: https://github.com/timvandam/eslint-bug/tree/main/src

I am not sure how eslint works internally, but this could be an issue with @typescript-eslint/parser

The repo has typescript + eslint setup and this `bug.ts`:
```ts
let x: number;

function dangerousFunction(): number {
	if (Math.random() > 0.5) throw new Error('Boom')
	return 123
}

try {
	x = dangerousFunction()
} catch (e) {
	//
}
```
